tiledb/sm/cpp_api/array.h

```diff
@@ -1115,7 +1115,7 @@ class Array {
    * std::vector<int> data_a1(max_elements["a1"].second);
    *
    * // In sparse reads, coords are also fixed-sized attributes.
-   * std::vector<int> coords(max_elements[TILEDB_COORDS].second);
+   * std::vector<int> coords(max_elements["__coords"].second);
```

Comment:
Use `constants::coords` instead of `"__coords"` in all places?

Reply:
I intentionally chose not do this because the files in `cpp_api` do not #include anything from `tiledb/sm/*`. To use `constants::coords`, we would need to break our existing abstraction that the cpp-api only uses the c-api.

Reply:
For reference, 44 uses of the core headers in the c-api but 0 in the cpp-api:
```
joe@joes-MBP % fgrep -r "include \"tiledb/sm" c_api | wc -l
      44
joe@joes-MBP % fgrep -r "include \"tiledb/sm" cpp_api | wc -l
       0
```

Reply:
Hmmm, this might bite us in the future if we change the coords name. I don't foresee us doing that, just noting down that it is a bit risky. There will not even be a compilation error in that case, so we are hoping to catch it in unit tests instead.

Reply:
This is true. To avoid this, I had originally just removed the deprecated tag in #1764. That is the only other reasonable approach that I can think of. Would you prefer to use that patch instead?

Reply:
No, the deprecation is valid too. Let's roll with this.

Reply:
Sounds good to me, thanks.